torch/nn/modules/transformer.py

```diff
@@ -466,15 +466,14 @@ def forward(self, src: Tensor, src_mask: Optional[Tensor] = None,
             why_not_sparsity_fast_path = "activation_relu_or_gelu was not True"
         elif not (self.norm1.eps == self.norm2.eps):
             why_not_sparsity_fast_path = "norm1.eps is not equal to norm2.eps"
-        elif src_mask is not None:
-            why_not_sparsity_fast_path = "src_mask is not supported for fastpath"
-        elif src.is_nested and src_key_padding_mask is not None:
-            why_not_sparsity_fast_path = "src_key_padding_mask is not supported with NestedTensor input for fastpath"
+        elif src.is_nested and (src_key_padding_mask is not None or src_mask is not None):
+            why_not_sparsity_fast_path = "src_key_padding_mask and src_mask are not supported with NestedTensor input"
```

Comment:
Is the message clear?  i.e., that either one causes this to fail.  I am hesitant to put too much time in error message engineering but helping users get to fastpath is important

Reply:
I think it can be made clearer, e.g. "supplying both src_key_padding_mask and src_mask is not supported with NestedTensor input" or "with NestedTensor input, only one of src_key_padding_mask and src_mask can be provided". Since in the next PR I'll working on making 4D inputs possible from Python level and modifying this code, I can as well adjust this message